### PR TITLE
feat(shareable-lists): bulk add items 

### PR DESCRIPTION
**Pull request by kschelonka**
Bulk add to an existing shareable list.

New creation is easier and will follow.

[POCKET-9504]

[POCKET-9504]: https://mozilla-hub.atlassian.net/browse/POCKET-9504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: servers/shareable-lists-api/package.json [hunk 1]
```diff
@@ -48,7 +51,8 @@
     "jest": "29.7.0",
     "jest-extended": "4.0.2",
     "nodemon": "3.0.3",
-    "prisma": "^4.12.0",
+    "prisma": "5.8.1",
```
Comment (bassrock) [inline servers/shareable-lists-api/package.json:54]: Yay!

File: servers/shareable-lists-api/schema-public.graphql [hunk 2]
```diff
@@ -233,4 +248,12 @@ type Mutation {
   Deletes a Shareable List Item. HIDDEN Lists cannot have their items deleted.
   """
   deleteShareableListItem(externalId: ID!): ShareableListItem!
+
+  """
+  Add a batch of items to an existing shareable list.
+  """
+  addToShareableList(
+    listExternalId: ID!
+    items: [AddItemInput!]! @constraint(minItems: 1, maxItems: 30)
```
Comment (bassrock) [inline servers/shareable-lists-api/schema-public.graphql:257]: 😍  on that constraint directive.

File: servers/shareable-lists-api/src/admin/resolvers/index.ts [hunk 3]
```diff
@@ -2,15 +2,15 @@ import { PocketDefaultScalars } from '@pocket-tools/apollo-utils';
...
-import { PrismaBigIntResolver } from '../../shared/resolvers/fields/PrismaBigInt';
+import { ShareableListItem } from '../../database/types';
 
 export const resolvers = {
   ...PocketDefaultScalars,
   ShareableListComplete: {
     user: UserResolver,
   },
   ShareableListItem: {
-    itemId: PrismaBigIntResolver,
```
Comment (bassrock) [inline servers/shareable-lists-api/src/admin/resolvers/index.ts:13]: Any idea why we needed this?

File: servers/shareable-lists-api/src/database/mutations/ShareableList.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import {
   NotFoundError,
   UserInputError,
 } from '@pocket-tools/apollo-utils';
-import { Visibility, ModerationStatus, PrismaClient } from '.prisma/client';
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/mutations/ShareableList.ts:6]: BYEEEE

File: servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts [hunk 5]
```diff
@@ -51,6 +75,8 @@ export async function createShareableListItem(
   }
 
   // check if an item with this URL already exists in this list
+  // TODO (@kschelonka): consider uniqeuness constraint over listId and url,
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts:78]: i like adding the names to the todos, good way to find yours

File: servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts [hunk 6]
```diff
@@ -104,18 +130,153 @@ export async function createShareableListItem(
...
+      userId,
+      moderationStatus: ModerationStatus.VISIBLE,
+    },
+  });
+  if (!list) {
+    throw new NotFoundError(
+      `A list with the ID of "${listExternalId}" does not exist`,
+    );
+  }
+  // TODO: Kat - fix this input type, it's not the same as graphql schema
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts:168]:
```suggestion
  // TODO: (@kschelonka) - fix this input type, it's not the same as graphql schema
```

File: servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts [hunk 7]
```diff
@@ -104,18 +130,153 @@ export async function createShareableListItem(
...
+        .executeTakeFirstOrThrow();
+      return { inserts, updatedList };
+    });
+  //send event bridge event for shareable-list-item-created event type
+  inserts.forEach((item) => {
+    if (item != null) {
+      sendEvent(EventBridgeEventType.SHAREABLE_LIST_ITEM_CREATED, {
+        // "bigint" headaches
+        // https://github.com/prisma/prisma/issues/7570
+        // Dealing with this properly is more work than the type safety is worth
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts:256]: 👍

File: servers/shareable-lists-api/src/database/queries/ShareableListItems.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  listId: number,
+): Promise<ListItemResponse[]> {
+  const { conn } = context;
+  const listItems = conn
+    .selectFrom('ListItem')
+    .where('listId', '=', listId)
+    .select(ListItemSelect)
+    .orderBy('sortOrder', 'asc')
+    .orderBy('createdAt', 'asc')
+    // Temporary tiebreaker sort for deterministic response
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/queries/ShareableListItems.ts:18]: Do we need to add a new field?

**Review by bassrock — APPROVED** [review]
LGTM, like the improvements in here too.

File: servers/shareable-lists-api/src/admin/resolvers/index.ts (hunk 3 above)
Comment (kschelonka) [reply servers/shareable-lists-api/src/admin/resolvers/index.ts:13]: I wonder if it was a schema change? I'll push to dev first

File: servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts (hunk 6 above)
Comment (kschelonka) [reply servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts:168]: Thank you, this I should have fixed/removed

**Pull request merged**